src/backtrace/unwind.rs

```diff
@@ -89,31 +96,36 @@ pub(crate) fn target(
 
         // If the frame didn't move, and the program counter didn't change, bail out (otherwise we
         // might print the same frame over and over).
-        corrupted = !cfa_changed && !program_counter_changed;
+        output.corrupted = !cfa_changed && !program_counter_changed;
 
-        if corrupted {
+        if output.corrupted {
             break;
         }
 
         if exception_entry {
-            raw_frames.push(RawFrame::Exception);
+            output.raw_frames.push(RawFrame::Exception);
 
             let fpu = match lr {
                 0xFFFFFFF1 | 0xFFFFFFF9 | 0xFFFFFFFD => false,
                 0xFFFFFFE1 | 0xFFFFFFE9 | 0xFFFFFFED => true,
-                _ => bail!("LR contains invalid EXC_RETURN value {:#010X}", lr),
+                _ => {
+                    output.processing_error =
+                        Some(anyhow!("LR contains invalid EXC_RETURN value {:#010X}", lr));
```

Comment:
(not necessarily for this PR): Since we use `output.processing_error = Some(anyhow!(...)` in multiple places we could have a small helper method for it 😄

Reply:
agreed, but I think this PR has overstayed its welcome :)